packages/mom5/package.py

```diff
@@ -5,114 +5,177 @@
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
 from spack.package import install, join_path, mkdirp
+from spack.build_systems import cmake, makefile
 from spack.version.version_types import GitVersion, StandardVersion
 
-# https://spack.readthedocs.io/en/latest/build_systems/makefilepackage.html
-class Mom5(MakefilePackage):
+class Mom5(CMakePackage, MakefilePackage):
     """MOM is a numerical ocean model based on the hydrostatic primitive equations."""
 
     homepage = "https://www.access-nri.org.au"
     git = "https://github.com/ACCESS-NRI/mom5.git"
-    submodules = True
 
-    maintainers("harshula", "penguian")
+    maintainers("dougiesquire", "harshula", "penguian")
 
+    # https://github.com/ACCESS-NRI/MOM5#LGPL-3.0-1-ov-file
+    license("LGPL-3.0-only", checked_by="dougiesquire")
+
+    version("mom_solo", branch="master")
+    version("mom_sis", branch="master")
     version("access-om2", branch="master", preferred=True)
     version("legacy-access-om2-bgc", branch="master")
     version("access-esm1.5", branch="access-esm1.5")
     version("access-esm1.6", branch="master")
 
-    variant("restart_repro", default=True, description="Reproducible restart build.")
-
-    with when("@access-esm1.6,access-om2,legacy-access-om2-bgc"):
-        variant("deterministic",
-                default=False,
-                description="Deterministic build.")
-        variant("optimisation_report",
-                default=False,
-                description="Generate optimisation reports.")
+    # NOTE: @mom matches both mom_solo and mom_sis
+    build_system(
+        conditional("makefile", when="@access-om2,legacy-access-om2-bgc,access-esm1.5"),
+        conditional("cmake", when="@mom,access-om2,legacy-access-om2-bgc,access-esm1.6"),
+        default="cmake",
+    )
+
+    with when("build_system=cmake"):
+        variant("build_type", default="RelWithDebInfo",
+            description="CMake build type",
+            values=("Debug", "Release", "RelWithDebInfo")
+        )
+        variant("deterministic", default=False, description="Deterministic build")
+
+    with when("build_system=makefile"):
+        variant("restart_repro", default=True, description="Reproducible restart build.")
+        variant(
+            "deterministic",
+            default=False,
+            description="Deterministic build",
+            when="@access-om2,legacy-access-om2-bgc,access-esm1.6"
+        )
+        variant(
+            "optimisation_report",
+            default=False,
+            description="Generate optimisation reports",
+            when="@access-om2,legacy-access-om2-bgc,access-esm1.6"
+        )
 
-    with when("@access-om2,legacy-access-om2-bgc"):
+    with when("@mom,access-om2,legacy-access-om2-bgc,access-esm1.6"):
         depends_on("netcdf-c@4.7.4:")
         depends_on("netcdf-fortran@4.5.2:")
-        # Depend on virtual package "mpi".
```

Comment:
Please restore the comment "# Depend on virtual package "mpi"."

Reply:
Fixed in [45a1981](https://github.com/ACCESS-NRI/spack-packages/pull/238/commits/45a19817aee613637f25091560d4fa76b49a8623)